src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java

```diff
@@ -60,7 +65,17 @@ protected void buildImage(ServiceHub hub, ImageConfiguration imageConfig)
         autoPullBaseImage(hub, imageConfig);
 
         MojoParameters params = createMojoParameters();
-        hub.getBuildService().buildImage(imageConfig, params, checkForNocache(imageConfig));
+        hub.getBuildService().buildImage(imageConfig, params, checkForNocache(imageConfig), addBuildArgs());
+    }
+
+    private Map<String, String> addBuildArgs() {
+        HashMap<String, String> args = new HashMap<>();
+        String[] argPairs = buildArgs.split(",");
```

Comment:
I think that a bit too simplistic because of missing escape hanlding (think about values containing `,`). 

What do you think about the idea to allow properties of the form `-Ddocker.build.arg.<variable_1>=<value_1> -Ddocker.build.arg.<variable_2>=<value_2> ...`. Thats how we model maps in the property handler anyway.

Of course then simple injection wouldn't work here but one would need to look it up actively from the `project` as properties when looking up.